Source/RxSwift/filterMapAs.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  filterMapAs.swift
+//  RxSwiftExt
+//
+//  Created by Jérôme Alves on 21/02/2018.
+//  Copyright © 2018 RxSwiftCommunity. All rights reserved.
+//
+
+import RxSwift
+
+extension ObservableType {
+
+    /**
+     Takes a sequence of elements and returns a sequence of elements casted as given type, filtering out any non castable values.
+
+     - parameter type: A type whose matches the element type of the input sequence
+     - returns: An observable sequence containing the elements 
+     */
+    public func filterMap<R>(as type: R.Type) -> Observable<R> {
+        return self.filterMap {
+            guard let result = $0 as? R else { return .ignore }
+            return .map(result)
+        }
```

Comment:
There are many different implementation possible for this operator. I chose this one to explicitly match the `filterMap` in the name.

Other possible implementations: 
```swift
return self.filter { $0 is R }.map { $0 as! R }
```
```swift
return self.map { $0 as? R }.unwrap()
```
```swift
return self.flatMap { ($0 as? R).map(Observable.just) ?? .empty() }
```
And I'm pretty sure there are more...

I don't know if an implementation is more optimized than another. 
If it does, feel free to tell me it and I'll update the code.